cmd/pd-server/main.go

```diff
@@ -224,7 +225,36 @@ func start(cmd *cobra.Command, args []string, services ...string) {
 
 	var sig os.Signal
 	go func() {
-		sig = <-sc
+	SignalHandler:
+		for {
+			select {
+			case sig = <-sc:
+				log.Info("Received a signal", zap.String("signal", sig.String()))
+				if svr.GetMember().IsLeader() {
+					log.Info("Transferring leadership due to signal", zap.String("signal", sig.String()))
+					err := svr.GetMember().ResignEtcdLeader(context.TODO(), svr.Name(), "")
```

Comment:
1. The `ResignEtcdLeader` used here does not transfer to a specified leader, it just do the resign. Can it save the etcd election timeout and leader re-election time?

2. IIRC, PD leader transfer may take several minutes in some case, and TiDB Operator even sets a wait time of up to 1500 minutes. What should we do in that case?

3. Kubernetes sends a SIGTERM signal to the process and waits for a certain number of seconds (30 by default) for it to shut down gracefully. If it does not shut down in time, the process is then killed through SIGKILL. How should we handle this situation?

Reply:
I did not manage to get the PD leader transfer to take several minutes, when testing on my laptop. I will follow up with the developer team working on PD. I was just trying to see if triggering a Resign during shutdown would help, but apparently it is not the issue here.